src/services/PythonRequest.ts

```diff
@@ -0,0 +1,133 @@
+    let pythonHeaders: string;
+    let pythonData = '';
+    let pythonQueryParams = '';
+    let pythonCookies = '';
+
+    if (!fetchUrl.fetchUrl.startsWith('http')) {
+      const url = new URL(fetchUrl.fetchUrl, window.location.href);
+      pythonUrlLine = `${pythonUrlLine} "${url.href}"`;
+    } else {
+      pythonUrlLine = `${pythonUrlLine} "${fetchUrl.fetchUrl}"`;
+    }
+
+    if (pythonUrlLine) {
+      pythonUrlLine = `${pythonUrlLine}${NEW_LINE}`;
+    }
+
+    pythonHeaders = JSON.stringify(
+      groupParamsByKey(fetchHeaders.entries() as IterableIterator<[string, string]>),
+      undefined,
+      2,
+    );
+
+    if (pythonHeaders && pythonHeaders != '{}') {
+      pythonHeaders = `headers = ${pythonHeaders}${NEW_LINE}`;
+    }
+
+    if (fetchUrl.queryParams?.toString()) {
+      pythonQueryParams = JSON.stringify(
+        groupParamsByKey(fetchUrl.queryParams.entries()),
+        undefined,
+        2,
+      );
+      pythonQueryParams = `params = ${pythonQueryParams}${NEW_LINE}`;
+    }
+
+    if (fetchOptions.body instanceof URLSearchParams) {
+      pythonData = `payload = ${fetchOptions.body.toString()} ${NEW_LINE}`;
+    } else if (fetchOptions.body) {
+      if (fetchOptions.body instanceof Object) {
+        try {
+          pythonData = `payload = ${JSON.stringify(fetchOptions.body, undefined, 2)}`;
+        } catch (err) {
+          // Ignore.
+        }
+      }
+      if (!pythonData) {
+        pythonData = `payload = '${fetchOptions.body.replace(/'/g, "'\"'\"'")}'${NEW_LINE}`;
+      }
+    }
+
+    if (pythonData) {
+      pythonData = `${pythonData}${NEW_LINE}`;
+    }
+
+    if (fetchUrl.cookieParams?.toString()) {
+      pythonCookies = `cookies = ${JSON.stringify(
+        groupParamsByKey(fetchUrl.cookieParams?.entries()),
+        undefined,
+        2,
+      )}${NEW_LINE}`;
+    }
+
+    const params = [
+      `"${this.method.toUpperCase()}"`,
+      'url',
+      pythonData ? 'data=payload' : '',
+      pythonHeaders ? 'headers=headers' : '',
+      pythonQueryParams ? 'params=params' : '',
+      pythonCookies ? 'cookies=cookies' : '',
+    ];
+    pythonResponseLine += `${params.filter(Boolean).join(', ')})${NEW_LINE}`;
+
+    let logging = `print(response.status_code)${NEW_LINE}`;
+    logging += this.getResponseLogging(fetchHeaders.get('accept') || '');
+
+    return [
+      imports,
+      pythonUrlLine,
+      pythonHeaders,
+      pythonQueryParams,
+      pythonCookies,
+      pythonData,
+      pythonResponseLine,
+      logging,
+    ]
+      .filter(Boolean)
+      .join(NEW_LINE);
+  }
+
+  private getResponseLogging(acceptHeader: string) {
+    switch (acceptHeader) {
+      case MIME_TYPES.APPLICATION_JSON:
+        return 'print(response.json())';
+      case MIME_TYPES.APPLICATION_OCTET_STREAM:
+        return 'print(response.content)';
+      case MIME_TYPES.APPLICATION_FORM_URLENCODED:
+      default:
+        return 'print(response.text)';
+    }
```

Comment:
will the requests be made only with these? no xml requests ?

Reply:
Yes, we didn't have any xml requests to test with and also for the requests generation we didn't do every use case since it will only be used by us (which is all basic GET requests).